File: app/src/main/java/com/lepitar/corvid19/SearchSchool/SchoolAdapter.java
package com.lepitar.corvid19.SearchSchool;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.lepitar.corvid19.MainActivity;
import com.lepitar.corvid19.R;
import com.lepitar.corvid19.SmsActivity;

import org.w3c.dom.Text;

import java.util.ArrayList;

import static android.content.Context.MODE_PRIVATE;

public class SchoolAdapter extends RecyclerView.Adapter<SchoolAdapter.CustomViewHolder> {

    private ArrayList<SchoolData> schoolData = new ArrayList<>();

    public SchoolAdapter(ArrayList<SchoolData> schoolData) {
        this.schoolData = schoolData;
    }

    @NonNull
    @Override
    public SchoolAdapter.CustomViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.schulname_list,parent, false);
        CustomViewHolder holder = new CustomViewHolder(view);
        return holder;
    }

    @Override
    public void onBindViewHolder(@NonNull final SchoolAdapter.CustomViewHolder holder, int position) {
        holder.schoolname.setText(schoolData.get(position).getSchoolName());
        holder.schoolname.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Context context = v.getContext();
                SharedPreferences sharedPreferences = context.getSharedPreferences("school", MODE_PRIVATE);
                SharedPreferences.Editor editor = sharedPreferences.edit();
                String schoolName = holder.schoolname.getText().toString();
                editor.putString("schoolName", schoolName);
                editor.apply();
                context.startActivity(new Intent(v.getContext(), MainActivity.class).addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP));
            }
        });
    }

    @Override
    public int getItemCount() {
        return (null != schoolData ? schoolData.size() : 0);
    }

    public class CustomViewHolder extends RecyclerView.ViewHolder {

        protected TextView schoolname;

        public CustomViewHolder(@NonNull View itemView) {
            super(itemView);
            this.schoolname = (TextView) itemView.findViewById(R.id.tv_schoolname);
        }
    }
}

File: app/src/main/java/com/lepitar/corvid19/SearchSchool/SchoolAdapter.kt
package com.lepitar.corvid19.SearchSchool

import android.app.Activity
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.lepitar.corvid19.ui.MainActivity
import com.lepitar.corvid19.R
import java.util.*

class SchoolAdapter(schoolData: ArrayList<SchoolData>?) : RecyclerView.Adapter<SchoolAdapter.CustomViewHolder>() {
    private var schoolData: ArrayList<SchoolData>? = ArrayList()
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CustomViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.schulname_list, parent, false)
        return CustomViewHolder(view)
    }

    override fun onBindViewHolder(holder: CustomViewHolder, position: Int) {
        holder.schoolname.text = schoolData!![position].schoolName
        holder.list_item.setOnClickListener { v ->
            val context = v.context
            var intent = Intent().apply {
                putExtra("name", schoolData!![position].schoolName)
                putExtra("schulCode", schoolData!![position].schulCode)
            }
            (context as Activity).setResult(Activity.RESULT_OK, intent)
            context.finish()
        }
    }

    override fun getItemCount(): Int {
        return schoolData?.size ?: 0
    }

    inner class CustomViewHolder(itemView: View) : ViewHolder(itemView) {
        val schoolname = itemView.findViewById<View>(R.id.tv_schoolname) as TextView
        val list_item = itemView.findViewById<View>(R.id.list_item) as LinearLayout
    }

    init {
        this.schoolData = schoolData
    }
}